ros2_batch_job/__main__.py

```diff
@@ -532,6 +532,10 @@ def need_package_from_pipy(pkg_name):
             else:
                 pip_packages += ["mypy==0.931"]
 
+        # We prefer to get pytest-timeout from the distribution if it exists.  If not we install it via pip.
+        if need_package_from_pipy("pytest-timeout"):
```

Comment:
This is annoying, but I'm pretty sure that to work, this needs to be:

```suggestion
        if need_package_from_pipy("pytest_timeout"):
```

Reply:
Is it different than the pip install instructions? `pip install pytest-timeout`

https://pypi.org/project/pytest-timeout/

Reply:
Yeah, the problem is that you can't do things like:

```
import pytest-timeout
```

as the Python interpreter things that is a minus sign.  In this particular case, `need_package_from_pipy` is using importlib with a string, but in my local testing this still doesn't work:

```
Python 3.10.2+ (main, Mar 14 2022, 05:18:40) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import importlib
>>> importlib.import_module('pytest-timeout')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'pytest-timeout'
>>> importlib.import_module('pytest_timeout')
<module 'pytest_timeout' from '/usr/lib/python3/dist-packages/pytest_timeout.py'>
>>> 
```

Reply:
Ah, I see. Thanks for the catch.

Reply:
Fixed in https://github.com/ros2/ci/pull/632/commits/472c21265116108b02d4f8930ee233bbf8117424.